src/graphql/resolvers/global-descriptor-resolvers.ts

```diff
@@ -0,0 +1,160 @@
+import { IResolvers } from 'graphql-tools';
+import services from '../../services';
+import { IContext } from '../context';
+import { PubSub } from 'graphql-subscriptions';
+import { GlobalDescriptorService } from '../../services/global-descriptor-service';
+
+export interface IDescriptor {
+	unit: String;
+	value: number;
+	longitude: number;
+	latitude: number;
+	unixTimestamp: number;
+}
+
+export enum TRANSACTION_TYPE {
+	TRANSACTION_HASH,
+	RECIEPT,
+	CONFIRMATION,
+	ERROR
+}
+/*
+interface IServiceMap {
+	globalDescriptorService: GlobalDescriptorService;
+	
+}
+class service{
+	 public serviceMap: IServiceMap = {
+		globalDescriptorService: new GlobalDescriptorService(),
+		
+	};
+	constructor(){
+
+	}
+
+	
+
+}
+*/
+
+const insertionSubscription = new PubSub();
+//const globalDescriptorService = new GlobalDescriptorService();
+const resolver: IResolvers = {
+	Query: {
+		getValuesForUnitGlobal: async(
+			_,
+			args: { unit: string },
+			context: IContext	
+		): Promise<IDescriptor[]> => {
+			if (context == undefined) {
+				Promise.reject('User context not available');
+			}
+			const ethAccId = context.getEtheriumAccountId();
+			return services.globalDescriptorService.getAllValuesRecordedForUnit(
+				ethAccId,
+				args.unit
+			);
+		},
+		getLatestUnitValueGlobal: async (
+			_,
+			args: { unit: string },
+			context: IContext
+		): Promise<number> => {
+			if (context == undefined) {
+				return Promise.reject('Global context not available');

```

Comment:
The context is the user-context. The auth token that the user passes has the user authentication information. Global context doesn't mean anything here.